### PR TITLE
First version of the 3D map view documentation

### LINKED ISSUE #2384: Please document the QGIS 3D config options
## Description

Please document the meaning of max tile resolution, max screen error, and max ground error in QGIS 3D config. This parameters are hard to understand why and why you should use them. The documentation should explain this.

## Checklist
- [x] I'm aware this repository is about QGIS Documentation and the issue I'm reporting is related to its usage.
- [ ] I have added a link to the page concerned by the issue

### LINKED ISSUE #2758: [feature] Allow moving camera up/down using page up/down keys
Original commit: https://github.com/qgis/QGIS/commit/2b5f46990c9151e459cc232eeca767f39f4bcfc9 by wonder-sk

Unfortunately this naughty coder did not write a description... :-(

### LINKED ISSUE #2759: [feature] Move camera keeping the position with Ctrl + arrow keys / mouse
Original commit: https://github.com/qgis/QGIS/commit/ef214d2ba1d4a1a55ad95c989056129994c9ff8b by wonder-sk

Unfortunately this naughty coder did not write a description... :-(

### PR DESCRIPTION
**Pull request by havatv**
The first version of the 3D Map View description in the user manual (Fixes #2384, Fixes #2758, Fixes #2759, ...).  Most of the text is taken from https://www.lutraconsulting.co.uk/blog/2018/03/01/working-with-qgis-3d-part-1/.
I hope we can just merge this, and then others can continue.

### CODE REVIEW

File: source/docs/user_manual/introduction/qgis_gui.rst [hunk 1]
```diff
@@ -632,6 +632,146 @@ view reflect changes you make in the legend area.
    also pan the map using the space bar or the click on mouse wheel: just move
    the mouse while holding down space bar or click on mouse wheel.
 
+
+.. index:: 3D Map view
+.. _`label_3dmapview`:
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/qgis_gui.rst]: could you add this reference to the table at https://docs.qgis.org/testing/en/docs/user_manual/introduction/qgis_gui.html#view

File: source/docs/user_manual/introduction/qgis_gui.rst [hunk 2]
```diff
@@ -632,6 +632,146 @@ view reflect changes you make in the legend area.
...
+
+The 3D map view has the same extent and view as the 2D canvas to start with.
+There is no dedicated toolbar for navigation in the 3D canvas.
+You zoom in/out and pan in the same way as in the main 2D canvas.
+You can also zoom in and out by dragging the mouse down/up with the
+right mouse button pressed.
+
+Navigation options for exploring the map in 3D:
+
+  * Tilt and rotate
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/qgis_gui.rst]: Should not be indented: align with "Navigation"
See https://github.com/qgis/QGIS-Documentation/blob/b6666bf88632fea0dee658aaa88c32779b92f6a7/source/docs/user_manual/introduction/qgis_gui.rst#3d-map-view

File: source/docs/user_manual/introduction/qgis_gui.rst [hunk 3]
```diff
@@ -632,6 +632,146 @@ view reflect changes you make in the legend area.
...
+
+      * Tilting the terrain (rotating it around a horizonal axis that goes
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/qgis_gui.rst]: Align with "Tilt"

File: source/docs/user_manual/introduction/qgis_gui.rst [hunk 4]
```diff
@@ -632,6 +632,146 @@ view reflect changes you make in the legend area.
...
+        through the center of the window) can be achieved by:
+
+          * Dragging the mouse forward / backward with the middle mouse button
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/qgis_gui.rst]: Align with "Tilting"

File: source/docs/user_manual/introduction/qgis_gui.rst [hunk 5]
```diff
@@ -632,6 +632,146 @@ view reflect changes you make in the legend area.
...
+        button pressed will change the camera angle corresponding to
+        directions of dragging
+      * Pressing :kbd:`Ctrl` and using the up/down/left/right keys turns
+        the camera upward, downward, to the left and to the right,
+        respectively
+
+  * Move the camera up / down
+
+      * Pressing the :kbd:`Page Up` / :kbd:`Page Down` keys moves the
+        terrain up and down, respectively
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/qgis_gui.rst]: if it's a single sub-item, no need to format as list and add it to the upper level, ie : `Move the camera up / down: Pressing the :kbd:`Page Up` / :kbd:`Page Down` keys moves the terrain up and down, respectively`

File: source/docs/user_manual/introduction/qgis_gui.rst [hunk 6]
```diff
@@ -632,6 +632,146 @@ view reflect changes you make in the legend area.
...
+
+  * Zoom in and out
+      * Dragging the mouse with the right mouse button pressed will
+        zoom in (backward dragging) and out (forward dragging)
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/qgis_gui.rst]: same as above

File: source/docs/user_manual/introduction/qgis_gui.rst [hunk 7]
```diff
@@ -632,6 +632,146 @@ view reflect changes you make in the legend area.
...
+
+  * Move the terrain around
+
+      * Dragging the mouse with the left mouse button pressed moves the
+        terrain around
+      * Using the up/down/left/right keys moves the
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/qgis_gui.rst]: `up/down/left/right` in many cases above there was space around / character. Personally, I prefer without spacing. Anyway, would be good to align.

File: source/docs/user_manual/introduction/qgis_gui.rst [hunk 8]
```diff
@@ -632,6 +632,146 @@ view reflect changes you make in the legend area.
...
+Before diving into the details, it is worth noting that terrain in 3D view
+is represented by a hierarchy of terrain tiles and as the camera moves
+closer to the terrain, existing tiles that do not have sufficient detail
+are replaced by smaller tiles with more details.
+Each tile has mesh geometry derived from the elevation raster layer and
+texture from 2D map layers.
+
+Configuration options and their meaning:
+
+  * **Elevation**: Raster to be used for generation of terrain.
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/qgis_gui.rst]: No indentation.
Also, should these not be guilabel?

File: source/docs/user_manual/introduction/qgis_gui.rst [hunk 9]
```diff
@@ -632,6 +632,146 @@ view reflect changes you make in the legend area.
...
+        
+To reset the camera view, click the |zoomFullExtent| button on the top of
+the 3D canvas panel.
+
+Terrain Configuration
+---------------------
+
+A terrain raster provides the elevation.
+Such a raster layer must contain a band that represents elevation.
+To select the terrain raster, click the Configure button on the top of
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/qgis_gui.rst]: Move it to a steps format, ie (**not a suggestion to commit** as I don't fix all the other lines and it has comments inline):
```suggestion
To select the terrain raster:

 #. Click the |**newIconToInsert**| :sup:`Configure` button on the top of the 3D canvas panel to open the :guilabel:`3D configuration` window
 #. Select the terrain raster layer in the :guilabel:`Elevation` pull-down menu
 (not sure of this) #. You are ready to go
```

File: source/docs/user_manual/introduction/qgis_gui.rst [hunk 10]
```diff
@@ -632,6 +632,146 @@ view reflect changes you make in the legend area.
...
+    Lower numbers mean more details in the scene at the expense of
+    increased rendering complexity.
+  * **Max. ground error**: The resolution of the terrain tiles at which
+    dividing tiles into more detailed ones will stop (splitting them
+    would not introduce any extra detail anyway).
+    This value limits the depth of the hierarchy of tiles: Lower values 
+    make the hierarchy deep, increasing rendering complexity.
+  * **Zoom levels**: Shows the number of zoom levels (depends on the
+    map tile resolution and max. ground error).
+  * **Show labels**: Toggles map labels on/off
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/qgis_gui.rst]: Can you add the |unchecked| substitution to these last options?

ref:
 ```
.. |unchecked| image:: /static/common/checkbox_unchecked.png
    :width: 1.3em
```

File: source/docs/user_manual/introduction/qgis_gui.rst [hunk 11]
```diff
@@ -632,6 +632,146 @@ view reflect changes you make in the legend area.
...
+  * **Show map tile info**: Include border and tile numbers for the
+    terrain tiles (useful for troubleshooting terrain issues)
+  * **Show bounding boxes**: Show 3D bounding boxes of the terrain
+    tiles (useful for troubleshooting terrain issues)
+
```
Comment (DelazJ) [inline source/docs/user_manual/introduction/qgis_gui.rst]: For reference, there's also "Show camera's view center" option (see https://github.com/qgis/QGIS/pull/7319)

File: source/docs/user_manual/introduction/qgis_gui.rst [hunk 12]
```diff
@@ -632,6 +632,115 @@ view reflect changes you make in the legend area.
...
+The 3D map view has the same extent and view as the 2D canvas to start with.
+There is no dedicated toolbar for navigation in the 3D canvas.
+You zoom in/out and pan in the same way as in the main 2D canvas.
+You can also zoom in and out by dragging the mouse down/up with the
+right mouse button pressed.
+
+Some additional options help you to explore the map in 3D:
+
+  * Tilt / rotate camera
```
[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/user_manual/introduction/qgis_gui.rst]: Fixed.

File: source/docs/user_manual/introduction/qgis_gui.rst [hunk 13]
```diff
@@ -632,6 +632,115 @@ view reflect changes you make in the legend area.
...
+        button pressed
+      * By pressing :kbd:`Ctrl` and using up/down/left/right keys
+
+  * Move the terrain
+  
+      * By pressing :kbd:`Alt` and dragging the mouse with the left mouse
+        button pressed
+      * By pressing :kbd:`Alt` and using up/down/left/right keys
+
+To reset the camera view, click Zoom Full button on the top of the 3D
```
[the remark this answers is not in the document]
Comment (havatv) [reply source/docs/user_manual/introduction/qgis_gui.rst]: Fixed.

File: source/docs/user_manual/introduction/qgis_gui.rst (hunk 1 above)
Comment (havatv) [reply source/docs/user_manual/introduction/qgis_gui.rst]: I am a bit worried that I might ruin the table.  Will try later.

File: source/docs/user_manual/introduction/qgis_gui.rst (hunk 2 above)
Comment (havatv) [reply source/docs/user_manual/introduction/qgis_gui.rst]: Fixed

File: source/docs/user_manual/introduction/qgis_gui.rst (hunk 3 above)
Comment (havatv) [reply source/docs/user_manual/introduction/qgis_gui.rst]: Fixed

File: source/docs/user_manual/introduction/qgis_gui.rst (hunk 4 above)
Comment (havatv) [reply source/docs/user_manual/introduction/qgis_gui.rst]: Fixed

File: source/docs/user_manual/introduction/qgis_gui.rst (hunk 7 above)
Comment (havatv) [reply source/docs/user_manual/introduction/qgis_gui.rst]: Added spaces.

File: source/docs/user_manual/introduction/qgis_gui.rst (hunk 8 above)
Comment (havatv) [reply source/docs/user_manual/introduction/qgis_gui.rst]: Fixed

File: source/docs/user_manual/introduction/qgis_gui.rst (hunk 9 above)
Comment (havatv) [reply source/docs/user_manual/introduction/qgis_gui.rst]: Fixed

File: source/docs/user_manual/introduction/qgis_gui.rst (hunk 10 above)
Comment (havatv) [reply source/docs/user_manual/introduction/qgis_gui.rst]: Fixed

File: source/docs/user_manual/introduction/qgis_gui.rst (hunk 11 above)
Comment (havatv) [reply source/docs/user_manual/introduction/qgis_gui.rst]: Fixed

File: source/docs/user_manual/introduction/qgis_gui.rst (hunk 1 above)
Comment (havatv) [reply source/docs/user_manual/introduction/qgis_gui.rst]: Fixed, I hope.

File: source/docs/user_manual/introduction/qgis_gui.rst (hunk 5 above)
Comment (havatv) [reply source/docs/user_manual/introduction/qgis_gui.rst]: I agree that in general it is weird, but in this case, I think it helps the user to find the actual keyboard / mouse combinations.

File: source/docs/user_manual/introduction/qgis_gui.rst (hunk 9 above)
Comment (havatv) [reply source/docs/user_manual/introduction/qgis_gui.rst]: I have tried to include the icon for the button now.  I still think look upon it is a button...

Comment (DelazJ) [reply source/docs/user_manual/introduction/qgis_gui.rst]: Sorry if you misunderstood me; I did not say it's not a button. My point is that there is no button labeled "Configure" but a button shown as icon (with a tooltip "Configure"), hence the suggestion *|icon| :sup:`Configure` button*.
I'm afraid it still misses the reference in substitutions.txt (and substitutions.rst) as per the [guidelines](https://docs.qgis.org/testing/en/docs/documentation_guidelines/writing.html#replacement). These ease life for people that would need to use that icon later.

Comment (havatv) [reply source/docs/user_manual/introduction/qgis_gui.rst]: Thanks for the help and patience! I have now updated substitutions.txt. I was not able to find a natural place for this substitution in substitutions.rst. Do you have a suggestion?

Comment (DelazJ) [reply source/docs/user_manual/introduction/qgis_gui.rst]: > Thanks for the help and patience!

You are welcome. If I can share the knowledge...
> Do you have a suggestion?

As [other basic icon](https://docs.qgis.org/testing/en/docs/documentation_guidelines/substitutions.html#other-basic-icons) or [map-navigation](https://docs.qgis.org/testing/en/docs/documentation_guidelines/substitutions.html#map-navigation-and-attributes)?

Comment (havatv) [reply source/docs/user_manual/introduction/qgis_gui.rst]: Other basic icons

**Comment by DelazJ** [comment]
great! Thanks  @havatv

**Pull request merged**